Nodejs/Product/Nodejs/Workspace/PackageJsonTestContainer.cs

```diff
@@ -0,0 +1,78 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.IO;
+using System.Threading;
+using Microsoft.VisualStudio.TestPlatform.ObjectModel;
+using Microsoft.VisualStudio.TestWindow.Extensibility;
+using Microsoft.VisualStudio.TestWindow.Extensibility.Model;
+
+namespace Microsoft.NodejsTools.Workspace
+{
+    public sealed class PackageJsonTestContainer : ITestContainer
+    {
+        private int version;
+
+        public PackageJsonTestContainer(ITestContainerDiscoverer discoverer, string source, string testRoot)
+        {
+            this.Discoverer = discoverer;
+            this.Source = source;
+            this.TestRoot = testRoot;
+        }
+
+        private PackageJsonTestContainer(PackageJsonTestContainer other) :
+            this(other.Discoverer, other.Source, other.TestRoot)
+        {
+            this.version = other.version;
+        }
+
+        public ITestContainerDiscoverer Discoverer { get; }
+
+        public string Source { get; }
+
+        public string TestRoot { get; }
+
+        public IEnumerable<Guid> DebugEngines { get { yield break; } }
```

Comment:
>yield break; [](start = 54, length = 12)

Unless there's an optimization I don't know about, this is way more expensive than `Array.Empty`.